src/odemis/acq/milling/tasks.py

```diff
@@ -0,0 +1,277 @@
+from odemis.acq.milling.patterns import (
+    MillingPatternParameters,
+    RectanglePatternParameters,
+    pattern_generator,
+)
+from odemis.acq.stream import FIBStream
+from odemis.acq.drift import align_reference_image
+
+class MillingSettings2:
+    """Represents milling settings for a single milling task"""
+
+    def __init__(self, current: float, voltage: float, field_of_view: float, mode: str = "Serial", channel: str = "ion", align: bool = True):
+        self.current = model.FloatContinuous(current, unit="A", range=(20e-12, 120e-9))
+        self.voltage = model.FloatContinuous(voltage, unit="V", range=(0, 30e3))
+        self.field_of_view = model.FloatContinuous(field_of_view, unit="m", range=(50e-06, 960e-06))
+        self.mode = model.StringEnumerated(mode, choices=set(["Serial", "Parallel"]))
+        self.channel = model.StringEnumerated(channel, choices=set(["ion"]))
+        self.align = model.BooleanVA(align) # align at the milling current
+
+    def to_json(self) -> dict:
+        return {"current": self.current.value,
+                "voltage": self.voltage.value,
+                "field_of_view": self.field_of_view.value,
+                "mode": self.mode.value,
+                "channel": self.channel.value,
+                "align": self.align.value}
+
+    @staticmethod
+    def from_json(data: dict) -> "MillingSettings2":
+        return MillingSettings2(current=data["current"],
+                                voltage=data["voltage"],
+                                field_of_view=data["field_of_view"],
+                                mode=data.get("mode", "Serial"),
+                                channel=data.get("channel", "ion"),
+                                align=data.get("align", True)
+                                )
+
+    def __repr__(self):
+        return f"{self.to_json()}"
+
+
+class MillingTaskSettings:
+    milling: MillingSettings2
+    patterns: List[MillingPatternParameters]
+
+    def __init__(self, milling: dict, patterns: List[MillingPatternParameters], name: str = "Milling Task"):
+        self.name = name
+        self.milling = milling
+        self.patterns = patterns
+
+    def to_json(self) -> dict:
+        return {"name": self.name, "milling": self.milling.to_json(), "patterns": [pattern.to_json() for pattern in self.patterns]}
+
+    @staticmethod
+    def from_json(data: dict):
+        return MillingTaskSettings(
+            name=data.get("name", "Milling Task"),
+            milling=MillingSettings2.from_json(data["milling"]),
+            patterns=[pattern_generator[p["pattern"]].from_json(p) for p in data["patterns"]])
+
+    def __repr__(self):
+        return f"{self.to_json()}"
+
+    def generate(self):
+        """Generate a list of milling patterns for the microscope"""
+        patterns = []
+        for pattern in self.patterns:
+            patterns.extend(pattern.generate())
+        return patterns
+
+class MillingTaskManager:
```

Comment:
shouldn't this be in the millmng module?

Reply:
ok, moved